internal/configuration/configuration.go

```diff
@@ -77,6 +77,15 @@ func WithDestinationsByName(d []ResourceConfig) Option {
 	}
 }
 
+// WithExtensionsByName is a Option that configures a configuration's
+// extensions.
```

Comment:
This is the same as `WithSourceByName` and `WithDestinationsByName`. At the end of the day, Extensions are just `model.ResourceConfiguration` types, just like sources and destinations.

The call to `withResourcesByName` takes Terraform's representation of a source, destination, extension and turns it into a `model.ResourceConfiguration`.